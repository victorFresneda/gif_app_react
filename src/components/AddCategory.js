import React, { useState } from 'react'
import PropTypes  from 'prop-types'        

const AddCategory = ({setCategories}) => {

    const [inptValue, setinptValue] = useState('');

    const handleInputChange = (e) => {
        setinptValue(e.target.value);
        console.log('llamado necesario')
    }

    const handleSubmit = (e) => {
        e.preventDefault();

        if( inptValue.trim().length > 2) {
            setCategories( cats => [inptValue, ...cats,]);
            setinptValue('');
        }
        
        
    }

    return (
        
        <form onSubmit={ handleSubmit }>
            <input 
                placeholder="Que buscas?"
                type="text"
                value={ inptValue}
                onChange={ handleInputChange }
            />
        </form>
    )
}

AddCategory.prototypes = { 
    setCategories: PropTypes.func.isRequired
}

export default AddCategory
