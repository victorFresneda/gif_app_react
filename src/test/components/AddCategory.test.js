import React from 'react';
import { shallow } from "enzyme"
import AddCategory from "../../components/AddCategory"


describe('Pruebas en < AddCategory/>', () => {
    
    const setCategories = jest.fn();
    let wrapper = shallow( < AddCategory setCategories={ setCategories } />);

    beforeEach( () => {
        jest.clearAllMocks();
        wrapper = shallow( < AddCategory setCategories={ setCategories } />);
    })
     
    test('debe de mostrarcse correctamente ', () => {
        
        expect(wrapper).toMatchSnapshot();
    })
        
    test('Debe de cambiar la caja de texto ', () => {
        
        const input = wrapper.find('input');
        const value = 'Cambio en el input';

        input.simulate('change', {target: { value }});
    })

    test('No debe de postear la informacion con sumit ', () => {
        
        wrapper.find('form').simulate('submit', { preventDefault(){} })

        expect( setCategories ).not.toHaveBeenCalled();
    })
    
    
    
})

