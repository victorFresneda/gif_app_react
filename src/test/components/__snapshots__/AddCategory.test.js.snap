// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en < AddCategory/> debe de mostrarcse correctamente  1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Que buscas?"
    type="text"
    value=""
  />
</form>
`;
